llama_index/node_parser/relational/markdown_element.py

```diff
@@ -0,0 +1,408 @@
+            table_id = element.id + "_table"
+            table_ref_id = element.id + "_table_ref"
+            # TODO: figure out what to do with columns
+            # NOTE: right now they're excluded from embedding
+            col_schema = "\n\n".join([str(col) for col in table_output.columns])
+            index_node = IndexNode(
+                text=str(table_output.summary),
+                metadata={"col_schema": col_schema},
+                excluded_embed_metadata_keys=["col_schema"],
+                id_=table_ref_id,
+                index_id=table_id,
+            )
+            table_str = table_df.to_string()
+            text_node = TextNode(
+                text=table_str,
+                id_=table_id,
+            )
+            nodes.extend([index_node, text_node])
+        else:
+            cur_text_el_buffer.append(str(element.element))
+    # flush text buffer
+    if len(cur_text_el_buffer) > 0:
+        cur_text_nodes = _get_nodes_from_buffer(cur_text_el_buffer, node_parser)
+        nodes.extend(cur_text_nodes)
+        cur_text_el_buffer = []
+
+    return nodes
+
+
+DEFAULT_SUMMARY_QUERY_STR = """\
+What is this table about? Give a very concise summary (imagine you are adding a caption), \
+and also output whether or not the table should be kept.\
+"""
+
+
+class MarkdownElementNodeParser(NodeParser):
+    """Markdown element node parser.
+
+    Splits a markdown document into Text Nodes and Index Nodes corresponding to embedded objects
+    (e.g. tables).
+
+    """
+
+    callback_manager: CallbackManager = Field(
+        default_factory=CallbackManager, exclude=True
+    )
+    llm: Optional[LLM] = Field(
+        default=None, description="LLM model to use for summarization."
+    )
+    summary_query_str: str = Field(
+        default=DEFAULT_SUMMARY_QUERY_STR,
+        description="Query string to use for summarization.",
+    )
+
+    def __init__(
+        self,
+        callback_manager: Optional[CallbackManager] = None,
+        llm: Optional[Any] = None,
+        summary_query_str: str = DEFAULT_SUMMARY_QUERY_STR,
+    ) -> None:
+        """Initialize."""
+        callback_manager = callback_manager or CallbackManager([])
+
+        return super().__init__(
+            callback_manager=callback_manager,
+            llm=llm,
+            summary_query_str=summary_query_str,
+        )
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "MarkdownElementNodeParser"
+
+    @classmethod
+    def from_defaults(
+        cls,
+        callback_manager: Optional[CallbackManager] = None,
+    ) -> "MarkdownElementNodeParser":
+        callback_manager = callback_manager or CallbackManager([])
+
+        return cls(
+            callback_manager=callback_manager,
+        )
+
+    def get_nodes_from_node(self, node: TextNode) -> List[BaseNode]:
+        """Get nodes from node."""
+        elements = extract_markdown_elements(
+            node.get_content(), table_filters=[filter_table]
```

Comment:
we probably want to use `node.get_content(metadata_mode=MetadataMode.NONE)` here

Reply:
I made the change, was under the impression that it was the default value, but apparently not for all type of Nodes

Reply:
Ah it is `NONE` by default lol my bad